snapcraft/internal/parser.py

```diff
@@ -233,6 +233,22 @@ def _process_wiki_entry(entry, master_parts_list):
         master_parts_list.update(parts_list)
 
 
+def _handle_wiki_entry(entry, master_parts_list):
+    wiki_errors = 0
+    try:
+        _process_wiki_entry(entry, master_parts_list)
+    except InvalidEntryError as e:
+        logger.warning('Invalid wiki entry: {!r}'.format(e))
+        wiki_errors = 1
+    except sources.MissingPackageError as e:
+        logger.warning(e)
+        # TODO: support multiple types of errors, see
+        # https://bugs.launchpad.net/snapcraft/+bug/1616613
+        wiki_errors = 1
+
+    return wiki_errors
```

Comment:
returning 1 or 0 to indicate error sounds like the job for an exception. You could just reraise it, and keep the try below:

try:
    _handle_wiki_entry
except SnapcraftError:
   wiki_errors += 1